### ISSUE TITLE
Document how to wrap an `eyre::Report` with a different error type

### ISSUE THREAD

**Issue by TheButlah** [yaahc/eyre#65]
Wrapping a source error type as the cause of an `eyre::Report` is very straightforward with the `wrap_err` function. But suppose I have an `eyre::Result<()>` and want to convert it into a `Result<(), ()>` without discarding the helpful backtrace/spantrace from `eyre`? Is that even possible?

If it is, can it be documented?

**Comment by yaahc** [comment]
> But suppose I have an `eyre::Result<()>` and want to convert it into a `Result<(), ()>` without discarding the helpful backtrace/spantrace from `eyre`? Is that even possible?

As stated, no, you definitely cannot convert an `eyre::Result<()>` to a `Result<(), ()>` without losing all relevant information. I think I may be misunderstanding what you mean, but based on what you wrote, you're essentially asking for `Result<(), eyre::Report> -> Result<(), ()>` which can be reduced to `eyre::Report -> ()`, aka converting `eyre::Report` to the [unit type](https://doc.rust-lang.org/stable/std/primitive.unit.html). This a `drop`, and can only be accomplished by discarding the original `eyre::Report`. All relevant information to an error such as backtrace and spantrace are stored in the `eyre::Report`, not the `Result`. We cannot discard this `Report` type without losing all of that helpful information.

**Comment by TheButlah** [comment]
> you're essentially asking for Result<(), eyre::Report> -> Result<(), ()> which can be reduced to eyre::Report -> (), aka converting eyre::Report to the unit type

Yeah that sums it up pretty well and it makes sense why that would not be viable. I guess what I'm really looking for is a way to continue to have whatever error type `E` my Result is supposed to use, but still propagate the helpful backtrace info that eyre contains. I can create a new `Report` but since its not a `Report<E>` I cannot access the wrapped error easily.

**Comment by yaahc** [comment]
I'm not sure I understand. Can you give me an example of what you're trying to do?

**Comment by TheButlah** [comment]
Wow I left this for a full year lol.

I believe what I was asking for was a way to propagate the information about the chain of underlying errors to other error error libs like thiserror.

I wanted to be able to use `eyre` in a library for prototyping easy errors, and then have a parent library that uses thiserror that handles error from all subsystems, some of which may be using `eyre`. But I didn't want to lose the helpful error chain.
